src/EFCore.PG/Query/Internal/NpgsqlSqlTranslatingExpressionVisitor.cs

```diff
@@ -56,17 +57,38 @@ public override SqlExpression TranslateCount(Expression expression = null)
                     _sqlExpressionFactory.Function("COUNT", new[] { _sqlExpressionFactory.Fragment("*") }, typeof(long))),
                 typeof(int), _sqlExpressionFactory.FindMapping(typeof(int)));
 
-//            // In PostgreSQL SUM() doesn't return the same type as its argument for smallint, int and bigint.
-//            // Cast to get the same type.
-//            // http://www.postgresql.org/docs/current/static/functions-aggregate.html
-//            if (sqlFunctionExpression.FunctionName == "SUM")
-//        {
-//            if (sqlFunctionExpression.Type == typeof(int))
-//                Sql.Append("::INT");
-//            else if (sqlFunctionExpression.Type == typeof(short))
-//                Sql.Append("::SMALLINT");
-//            return expr;
-//        }
+        // In PostgreSQL SUM() doesn't return the same type as its argument for smallint, int and bigint.
+        // Cast to get the same type.
+        // http://www.postgresql.org/docs/current/static/functions-aggregate.html
+        public override SqlExpression TranslateSum(Expression expression)
+        {
+            if (!(expression is SqlExpression sqlExpression))
+            {
+                sqlExpression = Translate(expression);
+            }
+
+            if (sqlExpression == null)
+            {
+                throw new InvalidOperationException(CoreStrings.TranslationFailed(expression.Print()));
+            }
```

Comment:
Braces. It's the influence of the EF team. Be strong , remove them and try to avoid them in the future. By the way, it's possible to simplify this multiline expression:

```csharp
var sqlExpression = expression as SqlExpression ?? Translate(expression) ?? throw new InvalidOperationException(CoreStrings.TranslationFailed(expression.Print()));
```

Reply:
Luckily you are here to keep an eye on me :)